src/vm/arm64/asmhelpers.asm

```diff
@@ -848,17 +851,96 @@ UM2MThunk_WrapperHelper_RegArgumentsSetup
     
     NESTED_END
 
+#ifdef FEATURE_HIJACK
+; ------------------------------------------------------------------
+; Hijack function for functions which return a reference type
+    NESTED_ENTRY OnHijackObjectTripThread
+    PROLOG_SAVE_REG_PAIR   fp, lr, #-112!
+    ; Spill callee saved registers 
+    PROLOG_SAVE_REG_PAIR   x19, x20, #16
+    PROLOG_SAVE_REG_PAIR   x21, x22, #32
+    PROLOG_SAVE_REG_PAIR   x23, x24, #48
+    PROLOG_SAVE_REG_PAIR   x25, x26, #64
+    PROLOG_SAVE_REG_PAIR   x27, x28, #80
+
+    str x0, [sp, #96]
+    mov x0, sp
+    bl OnHijackObjectWorker
+    ldr x0, [sp, #96]
+
+    EPILOG_RESTORE_REG_PAIR   x19, x20, #16
+    EPILOG_RESTORE_REG_PAIR   x21, x22, #32
+    EPILOG_RESTORE_REG_PAIR   x23, x24, #48
+    EPILOG_RESTORE_REG_PAIR   x25, x26, #64
+    EPILOG_RESTORE_REG_PAIR   x27, x28, #80
+    EPILOG_RESTORE_REG_PAIR   fp, lr,   #112!
+    EPILOG_RETURN
+    NESTED_END
+
+; ------------------------------------------------------------------
+; Hijack function for functions which return an interior pointer within an object allocated in managed heap
+    NESTED_ENTRY OnHijackInteriorPointerTripThread
+    PROLOG_SAVE_REG_PAIR   fp, lr, #-112!
+    ; Spill callee saved registers 
+    PROLOG_SAVE_REG_PAIR   x19, x20, #16
+    PROLOG_SAVE_REG_PAIR   x21, x22, #32
+    PROLOG_SAVE_REG_PAIR   x23, x24, #48
+    PROLOG_SAVE_REG_PAIR   x25, x26, #64
+    PROLOG_SAVE_REG_PAIR   x27, x28, #80
+
+    str x0, [sp, #96]
+    mov x0, sp
+    bl OnHijackInteriorPointerWorker
+    ldr x0, [sp, #96]
+
+    EPILOG_RESTORE_REG_PAIR   x19, x20, #16
+    EPILOG_RESTORE_REG_PAIR   x21, x22, #32
+    EPILOG_RESTORE_REG_PAIR   x23, x24, #48
+    EPILOG_RESTORE_REG_PAIR   x25, x26, #64
+    EPILOG_RESTORE_REG_PAIR   x27, x28, #80
+    EPILOG_RESTORE_REG_PAIR   fp, lr,   #112!
+    EPILOG_RETURN
+    NESTED_END
+
+; ------------------------------------------------------------------
+; Hijack function for functions which return a value type
+    NESTED_ENTRY OnHijackScalarTripThread
+    PROLOG_SAVE_REG_PAIR   fp, lr, #-112!
+    ; Spill callee saved registers 
+    PROLOG_SAVE_REG_PAIR   x19, x20, #16
+    PROLOG_SAVE_REG_PAIR   x21, x22, #32
+    PROLOG_SAVE_REG_PAIR   x23, x24, #48
+    PROLOG_SAVE_REG_PAIR   x25, x26, #64
+    PROLOG_SAVE_REG_PAIR   x27, x28, #80
+
+    str x0, [sp, #96]
+    ; ARM64TODO: HFA/HVA return value registers need to be saved
```

Comment:
You might as well just implement the HFA saves as well here:

HFA (Homogeneous Floating-point Aggregate) – this is a data type with 2-4 identical floating-point members (either floats or doubles)

So that would mean saving and restoring d0,d1,d2 and d3 here, instead of just d0